releasing/plugin-commands-publishing/src/pack.ts

```diff
@@ -122,7 +128,16 @@ export async function api (opts: PackOptions): Promise<PackResult> {
   if (!manifest.version) {
     throw new PnpmError('PACKAGE_VERSION_NOT_FOUND', `Package version is not defined in the ${manifestFileName}.`)
   }
-  const tarballName = `${manifest.name.replace('@', '').replace('/', '-')}-${manifest.version}.tgz`
+  let tarballName: string
+  const normalizedName = manifest.name.replace('@', '').replace('/', '-')
+  if (opts.filename) {
+    tarballName = opts.filename.replace('%s', normalizedName).replace('%v', manifest.version)
+    if (path.basename(tarballName) !== tarballName) {
+      throw new PnpmError('INVALID_FILENAME', 'Filename should not contain path specifiers. For specifying the directory where the tarball should be saved, use the --pack-destination option.')
+    }
```

Comment:
I think it is OK to use an `--out` flag instead of `--filename` (like Yarn). It is more convenient to declare both the directory path an filename in one flag. If both `--package-destination` and `--out` is set, an error can be thrown.